lib/aws-record/record/item_operations.rb

```diff
@@ -27,13 +28,29 @@ def self.included(sub_class)
       #
       # @raise [Aws::Record::Errors::KeyMissing] if a required key attribute
       #  does not have a value within this item instance.
-      def save
+      def save!
         dynamodb_client.put_item(
           table_name: self.class.table_name,
           item: build_item_for_save
         )
       end
 
+      # Attempts to save this record like #save!, but instead sets record as
+      # invalid. Matches style of ActiveRecord save and save! methods.
+      #
+      def save
+        save!
+      rescue Errors::KeyMissing => e
+        errors << e.message
+      end
+
+      # Is the record a valid record. True if #save was successful, false if
+      # Aws::Record::Errors::KeyMissing when using #save.
+      #
+      def valid?
+        errors.empty?
```

Comment:
To have this functionality, you're going to need to be able to flush the `errors` array on a successful save. Right now, an invalid write which is then corrected will still show invalid. Add a test for that case too (invalid write, correction, valid write, `#valid?` shows as true after second attempt).